packages/snap/src/keyring.ts

```diff
@@ -438,7 +460,9 @@ export class AccountAbstractionKeyring implements Keyring {
         16,
       )}`;
       if (!wallet.chains[chainId.toString()]) {
```

Comment:
I'm wondering, now that we use the caip-2 chain id in `wallet.chains` should we check for the caip-2 here? or only the `chainId` itself?

If yes, maybe we could have something like:

```typescript
const scope = toCaipChainId(CaipNamespaces.Eip155, chainId.toString());
if (!(scope in wallet.chains)) {
  wallet.chains[scope] = true;
}
```

Reply:
(or maybe use `Object.prototype.hasOwnProperty` like for `#doesAccountSupportChain`, if that's more idiomatic)

Reply:
GOOD CATCH [e1c4fcc](https://github.com/MetaMask/snap-account-abstraction-keyring/pull/51/commits/e1c4fccde0599a25233949a98cb70020ae7acf52)